### PR TITLE
rename a shadowed test and re-enable F811 to catch future cases (#8139)

### PR DESCRIPTION
**Pull request by alexmac**
(cherry picked from commit 3c0f1eb29d3512419ea65e7cdeb61ba3f3496f00)

## Checklist

- [ ] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [ ] Add a new news fragment into the `CHANGES/` folder
  * name it `<issue_or_pr_num>.<type>.rst` (e.g. `588.bugfix.rst`)
  * if you don't have an issue number, change it to the pull request
    number after creating the PR
    * `.bugfix`: A bug fix for something the maintainers deemed an
      improper undesired behavior that got corrected to match
      pre-agreed expectations.
    * `.feature`: A new behavior, public APIs. That sort of stuff.
    * `.deprecation`: A declaration of future API removals and breaking
      changes in behavior.
    * `.breaking`: When something public is removed in a breaking way.
      Could be deprecated in an earlier release.
    * `.doc`: Notable updates to the documentation structure or build
      process.
    * `.packaging`: Notes for downstreams about unobvious side effects
      and tooling. Changes in the test invocation considerations and
      runtime assumptions.
    * `.contrib`: Stuff that affects the contributor experience. e.g.
      Running tests, building the docs, setting up the development
      environment.
    * `.misc`: Changes that are hard to assign to any of the above
      categories.
  * Make sure to use full sentences with correct case and punctuation,
    for example:
    ```rst
    Fixed issue with non-ascii contents in doctest text files
    -- by :user:`contributor-gh-handle`.
    ```

    Use the past tense or the present tense a non-imperative mood,
    referring to what's changed compared to the last released version
    of this project.

### CODE REVIEW

**Comment by alexmac** [comment]
@Dreamsorcerer - here's one of the backports

File: tests/test_urldispatch.py [hunk 1]
```diff
@@ -1130,14 +1130,16 @@ def test_subapp_iter(app) -> None:
...
-
-def test_invalid_route_name(router) -> None:
+@pytest.mark.parametrize(
+    "route_name",
+    (
+        "invalid name",
+        "class",
+    ),
+)
+def test_invalid_route_name(router: Any, route_name: str) -> None:
```
Comment (Dreamsorcerer) [inline tests/test_urldispatch.py]:
```suggestion
def test_invalid_route_name(router, route_name: str) -> None:
```

**Comment by alexmac** [comment]
@Dreamsorcerer I've removed the other shadowed test (made the diff match master) - do you want the contrib file updated for this back port, and if so do I updated both the pr number and the text or just the text?

**Comment by Dreamsorcerer** [comment]
Ah, I think it's fine.

File: tests/test_resolver.py [hunk 2]
```diff
@@ -179,33 +179,7 @@ async def unknown_addrinfo(*args: Any, **kwargs: Any) -> List[Any]:
 
 
 async def test_close_for_threaded_resolver(loop) -> None:
-    resolver = ThreadedResolver(loop=loop)
```
Comment (webknjaz) [inline tests/test_resolver.py:182]: @Dreamsorcerer did you miss that this removes an unrelated test in a test module that the original commit didn't even touch?

Comment (webknjaz) [reply tests/test_resolver.py:182]: @alexmac it seems like the conflict resolution hasn't been completed correctly and this removed a legit test. Why?

Comment (webknjaz) [reply tests/test_resolver.py:182]: @alexmac was it exactly identical to the other test? It'd be useful to have in a PR description. And perhaps, do such things in separate PRs. Backports must never include changes that aren't directly related to resolving conflicts with the original patch.

Comment (Dreamsorcerer) [reply tests/test_resolver.py:182]: You can see the same test still exists, though the diff makes it a bit awkward to tell. The flake8 rule found this duplicate test that only existed on the 3.x branches. Could have been removed in another PR, but I don't think it makes any real difference.

Comment (webknjaz) [reply tests/test_resolver.py:182]: Yeah, I managed to find that. It'd be much more obvious (== less mental overhead) if this was explained explicitly.

As for a separate PR — this is the way. Patchback would also work to port from 3.10 to 3.9. The thing is that the `git cherry-pick -x` is supposed to contain a copy of another change, it even generates a reference `(cherry picked from commit xxxxxx)`. By making arbitrary changes, that makes it hard to track, breaking transparency of the updates to Git. Atomic commits/PRs is a convention that is designed around certain expectations/promises. Breaking them hurts future git paleontology, making the introspection harder. This effect is often not visible/obvious to the commit authors, making it a hard sell, but is quite important for keeping sane history.

**Pull request merged**